crates/events/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+/*!
+ */
+#![deny(missing_docs, missing_debug_implementations)]
+
+use wasm_bindgen::{JsCast, UnwrapThrowExt};
+use wasm_bindgen::closure::Closure;
+use web_sys::{Event, EventTarget};
+
+
+///
+pub struct EventListener<'a> {
+    target: EventTarget,
+    event_type: &'a str,
```

Comment:
When would the `event_type` not be `'static`?

Reply:
It allows you to dynamically choose the event type at runtime:

```rust
let s: String = get_string_somehow();

EventListener::new(&foo, &s, move |e| {
    // ...
})
```

Imagine `get_string_somehow()` returns a string at runtime (e.g. based on user input)

Although this is rather rare, it does happen!

As for `StaticEvent`... I'm not actually sure of any use cases right now, but since we cannot change it later, I figured it was best to make it more flexible (and consistent with `EventListener`).

If I can't think of any use cases, I'm fine with `StaticEvent` always using `'static` (though `EventListener` should still use `'a`)

Reply:
A better example than user input would be from JS code calling into Rust.

Reply:
Sure, there's lots of examples where dynamic events are useful. I only listed one of them.